react-accordion-component/my-app/src/Accordion.js

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import './Accordion.css';
+
+export default function Accordion({ topics }) {
+  const [current, setCurrent] = useState(null);
+
+  const lists = [];
+  for (let i = 0; i < topics.length; i++) {
+    lists.push(
+      <Panel
+        key={topics[i].id}
+        title={topics[i].title}
+        isActive={current === i}
+        onShow={handleClick}
+        content={topics[i].content}
+      />
+    );
+    function handleClick() {
+      if (current === topics[i].id) {
+        setCurrent(null);
+      } else {
+        setCurrent(topics[i].id);
+      }
+    }
+  }
```

Comment:
This is an interesting approach. It essentially creates a new handleClick function for each element in the list.

It seems odd, but it's actually no different than passing that in as an arrow function to your `onShow` prop:
`onShow={() => {
      if (current === topics[i].id) {
        setCurrent(null);
      } else {
        setCurrent(topics[i].id);
      }
}`
I probably wouldn't have reacted to this arrow function solution with "It seems odd." So your solution is just fine.

Another approach, though, is to put the handleClick function outside the for-loop and pass it the index:
`function handleClick(index)`
and
`onShow={() => handleClick(i)}`